env0/resource_team_project_assignment.go

```diff
@@ -59,14 +60,21 @@ func resourceTeamProjectAssignmentRead(ctx context.Context, d *schema.ResourceDa
 		return diag.Errorf("could not get TeamProjectAssignment: %v", err)
 	}
 
+	found := false
 	for _, assignment := range assignments {
 		if assignment.Id == id {
 			d.Set("project_id", assignment.ProjectId)
 			d.Set("team_id", assignment.TeamId)
 			d.Set("role", assignment.ProjectRole)
+			found = true
 			break
 		}
 	}
+	if !found {
+		if !d.IsNewResource() {
+			d.SetId("")
+		}
+	}
```

Comment:
```
if !d.IsNewResource() && !found {
		log.Printf("[WARN] Project (%s) not found, removing from state", d.Id())
		d.SetId("")
		return nil
	}
```

Looks better to me. I took it from [S3 bucket provider](https://github.com/hashicorp/terraform-provider-aws/blob/e86fca35f8f14709be928028a5550f29bbb21bc6/internal/service/s3/bucket.go#L830-L834) as example / reference.
Please do the same for others if can.

I am still not sure about this one. 
IIUC we should still `throw` on `d.IsNewResource()==true && someErrorOrBadCodeExists`.
I don't see any error returned on that case. @yaronya maybe you can revisit my thoughts here?

Reply:
i don't think we need to throw something, but we do need to return `nil` here

Reply:
Yes, @eranelbaz is correct - we don't need to throw anything. This is what we wanted to avoid in first place.
We should be removing the resource from state.

@samuel-br haven't we previously discussed it and said that we don't really need to check whether `IsNewResource` because TF doesn't `read` the resource before `create` (only before `update`/`delete`) so this would always return `false`?

Reply:
@yaronya i think `read ` is invoked on `create` too..

Reply:
`read` invokes on terraform refresh IIRC

Reply:
So I checked it with some test and @yaronya is right about `d.isNewResource()` but it still can be a factor on less common senarios as @eranelbaz said about refresh and maybe on `import `too (for now those resources dont have import but to be safe). 
So ill just change later for the changes @razbensimon  suggested

Reply:
the `if` is good now, just suggested adding logs to the user as I saw in the S3 bucket example.
for your decision @samuel-br

Reply:
"drift detected: TerraForm will remove {this resource} from state" is that good wording?

Reply:
"[WARN] Drift Detected: Terraform will remove {this resource} from state"

Reply:
c2ec119d609520f9038ee374e765173e65633ed5